### PR TITLE
fix conversation list parser in stern

### PR DESCRIPTION
**Pull request by battermann**
## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

File: tools/stern/src/Stern/API.hs [hunk 1]
```diff
@@ -402,15 +402,15 @@ getUserData uid = do
   convs <- Intra.getUserConversations uid
   clts <- Intra.getUserClients uid
   notfs <- Intra.getUserNotifications uid
-  consent <- Intra.getUserConsentValue uid
-  consentLog <- Intra.getUserConsentLog uid
+  consent <- (Intra.getUserConsentValue uid <&> Just) `catchE` \_ -> pure Nothing
```
Comment (fisx) [inline tools/stern/src/Stern/API.hs:405]: do these catch internal errors as well?  do we want that?

File: tools/stern/src/Stern/Intra.hs [hunk 2]
```diff
@@ -744,12 +746,12 @@ getUserConversations uid = do
...
             ( method GET
                 . header "Z-User" (toByteString' uid)
-                . path "/conversations"
+                . path "/v2/conversations"
                 . queryItem "size" (toByteString' batchSize)
                 . maybe id (queryItem "start" . toByteString') start
                 . expect2xx
             )
-      parseResponse (mkError status502 "bad-upstream") r
+      unVersioned @'V2 <$> parseResponse (mkError status502 "bad-upstream") r
```
Comment (fisx) [inline tools/stern/src/Stern/Intra.hs:754]: this wouldn't have happened if we used servant-client for calling internal api end-points...  :)

**Review by fisx — APPROVED** [review]

File: tools/stern/src/Stern/API.hs (hunk 1 above)
Comment (battermann) [reply tools/stern/src/Stern/API.hs:405]: I don't know, I just noticed that these fail and make the whole request fail, which is unnecessary, IMO.

Comment (battermann) [reply tools/stern/src/Stern/API.hs]: made a change, see comment below

**Comment by battermann** [comment]
Internally, stern makes multiple requests, some of which can fail and in this case the error will be shown in the response without making the overall request fail.
image

**Pull request merged**